### PR TITLE
Fix Cost API Bugs

### LINKED ISSUE #2143: Cost and Workspace Cost APIs uses default value for from_date and to_date separately causing unexpected behaviour
**Describe the bug**
Cost and Workspace Cost APIs uses default value for from_date and to_date separately causing unexpected behavior.

**Steps to reproduce**

1. Query GET /api/costs?from_date=2022-06-01T00%3A00%3A00&granularity=None
1. Should return valid costs data for current month but empty result set is returned.

**Expected result**
Should use current month for report if both from_date and to_date are empty
Should return 400 Bad Request if from_date is bigger than to_date
to_date should default to tomorrows date if not set
from_date should default to today's date if not set

### LINKED ISSUE #2144: Workspace existence exception is not handled in workspace costs api (throws EntityDoesNotExist)
**Describe the bug**
Workspace does not exists is not handled in workspace costs api (throws EntityDoesNotExist)

**Steps to reproduce**

1.Query GET /api/workspaces/123/costs
1. API returns EntityDoesNotExist 

**Expected result**
404 Not Found

### PR DESCRIPTION
**Pull request by guybartal**
# Fixes #2144 Fixes #2143

## How is this addressed

- if both ```from_date``` and ```to_date``` query params are ```None``` then query cost management API by ```MonthToDate```
- throw ```WorkspaceDoesNotExist``` exception if workspace is not found and handle in workspace costs API
- validate ```workspace_id``` as ```UUID4``` type

### CODE REVIEW

**Review by tamirkamara — COMMENTED** [review]
@guybartal shouldn't there be some tests that cover these bug/fixes?

File: api_app/api/routes/costs.py [hunk 1]
```diff
@@ -13,41 +15,36 @@
...
-    converted_datetime.replace(tzinfo=pytz.UTC)
-    return converted_datetime
+def validate_report_period(from_date: Optional[datetime], to_date: Optional[datetime]):
+    if from_date is None and to_date is None:
+        # valid option, month to date report
+        return
 
-
-def validate_report_period(from_date: datetime, to_date: datetime):
+    if to_date is None:
```
Comment (SharonHart) [inline api_app/api/routes/costs.py:29]: if to_date is None or from_date >= to_date

File: api_app/api/routes/costs.py [hunk 2]
```diff
@@ -13,41 +15,36 @@
...
 
 
 class CostsQueryParams:
     def __init__(
         self,
-        from_date: datetime = Query(default=get_first_day_of_month(),
-                                    description="The start date to pull data from, default value first day of month (iso-8601, UTC)."),
-        to_date: datetime = Query(default=get_tomorrow_datetime(),
-                                  description="The end date to pull data to, default tomorrow's date (iso-8601, UTC)."),
+        from_date: Optional[datetime] = Query(default=None, description="The start date to pull data from, requiered if to_date is set, otherwise report will return month to date (iso-8601, UTC)."),
```
Comment (SharonHart) [inline api_app/api/routes/costs.py:46]: default=None is redundant?

File: api_app/services/cost_service.py [hunk 3]
```diff
@@ -69,29 +73,32 @@ def query_tre_costs(self, tre_id, granularity: GranularityEnum, from_date: datet
...
-                                                  workspace_id),
-            workspace_services=self.__get_workspace_services_costs(granularity, query_result_dict,
-                                                                   workspace_services_repo,
-                                                                   user_resource_repo,
-                                                                   workspace_id)
-        ))
+        try:
+            workspace = workspace_repo.get_workspace_by_id(workspace_id)
+
+            workspace_cost_report: WorkspaceCostReport = WorkspaceCostReport(**dict(
```
Comment (SharonHart) [inline api_app/services/cost_service.py:88]: is `**dict` really necessary?

File: api_app/api/routes/costs.py (hunk 2 above)
Comment (guybartal) [reply api_app/api/routes/costs.py:46]: TypeError: Query() missing 1 required positional argument: 'default'

File: api_app/api/routes/costs.py [hunk 4]
```diff
@@ -72,13 +69,16 @@ async def costs(
...
     validate_report_period(params.from_date, params.to_date)
-    return cost_service.query_tre_workspace_costs(
-        workspace_id, params.granularity, params.from_date, params.to_date,
-        workspace_repo, workspace_services_repo, user_resource_repo)
+    try:
+        return cost_service.query_tre_workspace_costs(
+            str(workspace_id), params.granularity, params.from_date, params.to_date,
+            workspace_repo, workspace_services_repo, user_resource_repo)
+    except WorkspaceDoesNotExist:
+        raise HTTPException(status_code=status.HTTP_404_NOT_FOUND, detail=strings.WORKSPACE_DOES_NOT_EXIST)
```
[the remark this answers is not in the document]
Comment (guybartal) [reply api_app/api/routes/costs.py:84]: I think this is useful for cross app routes exception handling in oppose to specific route use case, do you agree?

**Comment by guybartal** [comment]
> @guybartal shouldn't there be some tests that cover these bug/fixes?

added unit tests to cover time frame for ```cost_service```

**Review by SharonHart — APPROVED** [review]

**Pull request merged**